acme/dns_challenge.go

```diff
@@ -144,7 +145,8 @@ func checkAuthoritativeNss(fqdn, value string, nameservers []string) (bool, erro
 	return true, nil
 }
 
-// dnsQuery sends a DNS query to the given nameserver.
+// dnsQuery will attempt 3 times, a query to the given nameserver.
+// If the given nameserver is `RecursiveNameserver`, it will attempt the `altRecursiveNameserver` on the 2nd and 3rd attempts.
 // The nameserver should include a port, to facilitate testing where we talk to a mock dns server.
 func dnsQuery(fqdn string, rtype uint16, nameserver string, recursive bool) (in *dns.Msg, err error) {
```

Comment:
> I guess, my final opinion is "This is a better way to do it, and therefore we should" - even if my short-sightedness doesn't necessarily appreciate its application ;)

Glad we are on the same page here (minus the short-sightedness) 😄 
rfc2136 is the only provider that passes a custom nameserver to `FindZoneByFqdn()`.

> In making this change, how about also adding either a variable retry, or set it to len(nameservers)+1?

Either is fine with me.